### PR TITLE
Remove mentions of datapack in favor of DataPack

### PR DESCRIPTION
**Pull request by moehreag**
Discussions of this:
<https://github.com/QuiltMC/quilt-mappings/pull/433#discussion_r1212221928>
<https://discord.com/channels/817576132726620200/1104011466660659273/1113554858432278559>

### CODE REVIEW

**Review by EnnuiL — CHANGES_REQUESTED** [review]
Don't forget about `MinecraftServer.reloadResources`! although huh, i'm surprised that it passed the lint

**Comment by moehreag** [comment]
> Don't forget about `MinecraftServer.reloadResources`! although huh, i'm surprised that it passed the lint

~~hm, I don't think lambda parameters are linted at all currently, but I'm not sure (since they also aren't counted in the total parameters stats (or any other stats))~~ I only just now managed to understand what you actually meant, and I have no idea.

Should the javadocs also use `dataPacks` instead of `datapacks` ? I assume so, but please tell me if that's not the case.

File: mappings/net/minecraft/server/MinecraftServer.mapping [hunk 1]
```diff
@@ -301,12 +301,12 @@ CLASS net/minecraft/server/MinecraftServer net/minecraft/server/MinecraftServer
 		ARG 1 serverPort
 	METHOD m_prhzfzen getDefaultGameMode ()Lnet/minecraft/unmapped/C_lghcpyvl;
 	METHOD m_pwdmbosg reloadResources (Ljava/util/Collection;)Ljava/util/concurrent/CompletableFuture;
-		COMMENT Reloads this server's datapacks.
+		COMMENT Reloads this server's dataPacks.
```
Comment (LambdAurora) [inline mappings/net/minecraft/server/MinecraftServer.mapping]:
```suggestion
		COMMENT Reloads this server's data-packs.
```

This is javadoc, not code...

**Review by LambdAurora — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/server/MinecraftServer.mapping (hunk 1 above)
Comment (moehreag) [reply mappings/net/minecraft/server/MinecraftServer.mapping]: I just asked whether this should be modified as well, but thanks for the clear response!

**Comment by eristhea** [comment]
> since they also aren't counted in the total parameters stats (or any other stats)

note that this is an enigma bug, we've fixed it for 1.8!

**Review by EnnuiL — APPROVED** [review]

File: mappings/net/minecraft/server/MinecraftServer.mapping [hunk 2]
```diff
@@ -301,12 +301,12 @@ CLASS net/minecraft/server/MinecraftServer net/minecraft/server/MinecraftServer
...
 	METHOD m_pwdmbosg reloadResources (Ljava/util/Collection;)Ljava/util/concurrent/CompletableFuture;
-		COMMENT Reloads this server's datapacks.
+		COMMENT Reloads this server's data-packs.
 		COMMENT
 		COMMENT @return a completable future which specifies whether the reload was successful
 		COMMENT A reload has failed when the future is exceptionally completed.
 		COMMENT @see CompletableFuture
-		ARG 1 datapacks
+		ARG 1 dataPacks
 			COMMENT a collection of datapacks to reload with
```
Comment (LambdAurora) [inline mappings/net/minecraft/server/MinecraftServer.mapping]:
```suggestion
			COMMENT a collection of data-packs to reload with
```

**Review by LambdAurora — APPROVED** [review]

**Review by supersaiyansubtlety — APPROVED** [review]

**Pull request merged**